tests/core/pyspec/eth2spec/test/phase0/unittests/fork_choice/test_on_block.py

```diff
@@ -317,7 +241,151 @@ def test_on_block_outside_safe_slots_but_finality(spec, state):
     block = build_empty_block_for_next_slot(spec, just_fin_state)
     signed_block = state_transition_and_sign_block(spec, deepcopy(just_fin_state), block)
 
+    # NOTE: Mock store so that the modified state could be accessed
+    parent_block = last_signed_block.message.copy()
+    parent_block.state_root = just_fin_state.hash_tree_root()
+    store.blocks[block.parent_root] = parent_block
+    store.block_states[block.parent_root] = just_fin_state.copy()
+    assert block.parent_root in store.blocks.keys()
+    assert block.parent_root in store.block_states.keys()
+
     run_on_block(spec, store, signed_block)
 
     assert store.finalized_checkpoint == new_finalized
     assert store.justified_checkpoint == new_justified
+
+
+@with_all_phases
+@spec_state_test
+def test_new_finalized_slot_is_not_justified_checkpoint_ancestor(spec, state):
+    """
+    J: Justified
+    F: Finalized
+    pre-store:
+        epoch
+        [0] <- [1] <- [2] <- [3] <- [4] <- [5]
+         F                    J
+
+    another_state:
+        [0] <- [1] <- [2] <- [3] <- [4] <- [5]
```

Comment:
Are "0" and "1" the same in both diagrams?

Reply:
In this case, `another_state` forked from `state`'s epoch 0.
I updated the docstring. 👍